fondant/compiler.py

```diff
@@ -0,0 +1,103 @@
+import json
+import logging
+import typing as t
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+import yaml
+
+from fondant.pipeline import ComponentOp, Pipeline
+
+logger = logging.getLogger(__name__)
+
+
+class Compiler(ABC):
+    """Abstract base class for a compiler."""
+
+    def __init__(self, pipeline: Pipeline):
+        self.pipeline = pipeline
+
+    @abstractmethod
+    def compile(self):
+        """Abstract method to invoke compilation."""
+
+
+class DockerCompiler(Compiler):
+    """Compiler that creates a docker-compose spec from a pipeline."""
+
+    def compile(self, package_path: str = "docker-compose.yml") -> None:
+        """Compile a pipeline to docker-compose spec that is saved on the package_path."""
+        self._patch_path()
+        spec = self._generate_spec()
+        with open(package_path, "w") as outfile:
+            yaml.safe_dump(spec, outfile)
+
+    @staticmethod
+    def _safe_component_name(component_name: str) -> str:
+        """Transform a component name to a docker-compose friendly one.
+        eg: `Component A` -> `component_a`.
+        """
+        return component_name.replace(" ", "_").lower()
+
+    def _patch_path(self):
+        """Helper that checks if the base_path is local or remote,
+        if local it patches the base_path and prepares a bind mount.
+        """
+        base_path = Path(self.pipeline.base_path)
+        # check if base path is an existing local folder
+        if base_path.exists():
```

Comment:
what if the path is local but does not exist? it will assume it's remote In this case

Reply:
Yes

Reply:
alright but maybe we can improve the error handling or try creating a new local path? now it assumes that if it's local and that it does not exist that it's remote (you assign it the base path)